src/sentry/incidents/endpoints/serializers.py

```diff
@@ -151,189 +348,110 @@ def update(self, instance, validated_data):
-                if not access.has_team(team):
-                    raise serializers.ValidationError("Team does not exist")
-            elif target_type == AlertRuleTriggerAction.TargetType.USER:
-                try:
-                    user = User.objects.get(id=identifier)
-                except User.DoesNotExist:
-                    raise serializers.ValidationError("User does not exist")
-
-                if not OrganizationMember.objects.filter(
-                    organization=self.context["organization"], user=user
-                ).exists():
-                    raise serializers.ValidationError("User does not belong to this organization")
-        elif attrs.get("type") == AlertRuleTriggerAction.Type.SLACK:
-            if not attrs.get("integration"):
-                raise serializers.ValidationError(
-                    {"integration": "Integration must be provided for slack"}
-                )
-
-        return attrs
-
-    def create(self, validated_data):
-        return create_alert_rule_trigger_action(trigger=self.context["trigger"], **validated_data)
+            print("Running create!")
+            # TODO: Remove this, just temporary while we're supporting both fields.
+            if "aggregation" not in validated_data:
+                raise serializers.ValidationError("aggregation is required")
 
-    def _remove_unchanged_fields(self, instance, validated_data):
-        changed = False
-        if (
-            validated_data.get("type", instance.type) == AlertRuleTriggerAction.Type.SLACK.value
-            and validated_data["target_identifier"] != instance.target_display
-        ):
-            changed = True
-        for field_name, value in list(six.iteritems(validated_data)):
-            # Remove any fields that haven't actually changed
-            if isinstance(value, Enum):
-                value = value.value
-            if getattr(instance, field_name) != value:
-                changed = True
-                break
-        return validated_data if changed else {}
+            print("saving! validated data:",validated_data)
+            triggers_data = validated_data.pop('triggers')
+            alert_rule = create_alert_rule(organization=self.context["organization"], **validated_data)
+            if alert_rule:
+                for trigger_data in triggers_data:
+                    trigger_action_data = trigger_data.pop("actions")
+
+                    trigger_serializer = AlertRuleTriggerSerializer(
+                        context={
+                            "organization": self.context["organization"],
+                            "alert_rule": alert_rule,
+                            "access": self.context["access"],
+                        },
+                        data=trigger_data,
+                    )
+
+                    if trigger_serializer.is_valid():  # AlertRuleTriggerSerializer does the validation
```

Comment:
If we get nested serializers working, it might make sense for the action data to be handled by the `AlertRuleTriggerSerializer`, since that data is more directly related there (if this is hard don't worry too much)

Reply:
Nested serializers are working :)

I implemented get_attrs in AlertRuleTriggerSerializer that adds the actions to the serializer. now the frontend can also use them in the list display, and we can possibly add a "name"/"label" field to the actions like the issue alert ones have.

Reply:
Hmm I wish the nested serializer could automatically call it's own update/create...and then I would also just drop the action_data loop into the AlertRuleTrigger's update/create. Might be a way.